### PR TITLE
PathPermissions and RolePermissions to follow JavaBean notation

### PR DESCRIPTION
**Pull request by gapra-msft**
Resolves https://github.com/Azure/azure-sdk-for-java/issues/6318

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathPermissions.java [hunk 1]
```diff
@@ -278,19 +278,19 @@ public PathPermissions other(RolePermissions other) {
...
 
     /**
      * Sets the value of the sticky bit. The sticky bit may be set on directories, the files in that
      * directory may only be renamed or deleted by the file's owner, the directory's owner, or the root user.
      *
-     * @param stickyBit {@code True} to set the sticky bit and {@code false} to clear it.
+     * @param setStickyBit {@code True} to set the sticky bit and {@code false} to clear it.
      * @return The updated PathPermissions object.
      */
-    public PathPermissions stickyBit(boolean stickyBit) {
```
Comment (jaschrep-msft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathPermissions.java]: is the argument supposed to be renamed like this?

Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathPermissions.java]: Sas is formatted like this 
/**
     * @return the read permission status
     */
    public boolean hasReadPermission() {
        return readPermission;
    }

    /**
     * Sets the read permission status.
     *
     * @param hasReadPermission Permission status to set
     * @return the updated BlobContainerSasPermission object
     */
    public BlobContainerSasPermission setReadPermission(boolean hasReadPermission) {
        this.readPermission = hasReadPermission;
        return this;
    }

Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathPermissions.java]: I think I should rename it to has

**Comment by gapra-msft** [comment]
> Confirm all these boolean arguments being labeled `set_____` is the desired naming, because that seems odd to me. Good otherwise.

This matches the standard JavaBean notation which I think we should be following for things like this, maybe Alan, Sima or Jonathan can provide further insight

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathPermissions.java [hunk 2]
```diff
@@ -301,11 +301,11 @@ public PathPermissions stickyBit(boolean stickyBit) {
      *
      * @return {@code true} if there is more information in the ACL. {@code false} otherwise.
      */
-    public boolean extendedInfoInAcl() {
+    public boolean hasExtendedInfoInAcl() {
         return this.extendedInfoInAcl;
     }
 
-    PathPermissions extendedInfoInAcl(boolean extendedInfoInAcl) {
+    PathPermissions setExtendedInfoInAcl(boolean extendedInfoInAcl) {
```
Comment (sima-zhu) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathPermissions.java]: Can you follow the rule above?

Comment (sima-zhu) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathPermissions.java]: `boolean hasExtendedInfoInAcl`

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java [hunk 3]
```diff
@@ -187,81 +173,75 @@ public boolean equals(Object o) {
...
+     * Sets the read permission status.
      *
-     * @param read {@code true} if read permissions should be granted.
-     * @return The updated RolePermissions object.
+     * @param setReadPermission Permission status to set
+     * @return the updated RolePermissions object
      */
-    public RolePermissions read(boolean read) {
-        this.read = read;
+    public RolePermissions setReadPermission(boolean setReadPermission) {
```
Comment (sima-zhu) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java]: boolean setReadPermission -> hasReadPermission

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java [hunk 4]
```diff
@@ -187,81 +173,75 @@ public boolean equals(Object o) {
...
+     * Sets the write permission status.
      *
-     * @param write {@code true} if write permissions should be granted.
-     * @return The updated RolePermissions object.
+     * @param setWritePermission Permission status to set
+     * @return the updated RolePermissions object
      */
-    public RolePermissions write(boolean write) {
-        this.write = write;
+    public RolePermissions setWritePermission(boolean setWritePermission) {
```
Comment (sima-zhu) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java]: Same above

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java [hunk 5]
```diff
@@ -187,81 +173,75 @@ public boolean equals(Object o) {
...
+     * Sets the execute permission status.
      *
-     * @param execute {@code true} if execute permissions should be granted.
-     * @return The updated RolePermissions object.
+     * @param setExecutePermission Permission status to set
+     * @return the updated RolePermissions object
      */
-    public RolePermissions execute(boolean execute) {
-        this.execute = execute;
+    public RolePermissions setExecutePermission(boolean setExecutePermission) {
```
Comment (sima-zhu) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java]: Same above

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java (hunk 3 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java]: I think setters should be set and getters should be has (this is how we do it for SASPermission)

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java (hunk 4 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java]: see above

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java (hunk 5 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java]: see above

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java (hunk 3 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java]: Ohhhh nvm

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java (hunk 5 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java]: done

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java (hunk 4 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java]: done

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java (hunk 3 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/RolePermissions.java]: done

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathPermissions.java (hunk 2 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/models/PathPermissions.java]: done

**Pull request merged**